.github/workflows/test.yml

```diff
@@ -4,10 +4,34 @@ on:
   push
 
 jobs:
-  build:
+
+  static_code_analysis:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - run: g++ -Wall main.cpp Unit.cpp Game.cpp
+
+      - run: sudo apt-get install cppcheck
+      - run: cppcheck --enable=all --error-exitcode=1 *.cpp 2> static-code-analysis.txt
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: Static_code_analysis_artifact
+          path: ./static-code-analysis.txt
+
+  build_test_and_memory_leak_check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - run: g++ -Wall -Wextra -Werror *.cpp
+
       - run: chmod +x test.sh
-      - run: ./test.sh
\ No newline at end of file
+      - run: ./test.sh
+
+      - run: sudo apt-get install valgrind
+      - run: valgrind --error-exitcode=1 ./a.out unit1.json unit2.json 2> ./memory-leak-check.txt
```

Comment:
A valgrindnak még egy `--leak-check=full` opciót is meg kell adni, különben nem minden hibát tekint errornak.